app/src/main/java/net/gsantner/opoc/ui/SearchEngine.java

```diff
@@ -0,0 +1,570 @@
+                if (_currentSearchDepth > _config._maxSearchDepth) {
+                    break;
+                }
+                _currentQueueLength = mainQueue.size() + 1;
+                publishProgress(_currentQueueLength, _currentSearchDepth, _result.size(), _countCheckedFiles);
+
+                mainQueue.addAll(currentDirectoryHandler(currentDirectory));
+            }
+
+            if (_isCanceled && _result.size() == 0) {
+                cancel(true);
+            }
+
+            return _result;
+        }
+
+
+        private Queue<File> currentDirectoryHandler(File currentDir) {
+            Queue<File> subQueue = new LinkedList<File>();
+
+            try {
+                if (!currentDir.canRead() || currentDir.isFile()) {
+                    return subQueue;
+                }
+
+                File[] subDirsOrFiles = currentDir.listFiles();
+                for (int i = 0; i < subDirsOrFiles.length && !isCancelled() && !_isCanceled; i++) {
+                    _countCheckedFiles++;
+                    File subDirOrFile = subDirsOrFiles[i];
+
+                    if (!subDirOrFile.canRead()) {
+                        continue;
+                    }
+
+                    if (subDirOrFile.isDirectory()) {
+                        File directory = subDirOrFile;
+                        if (isFolderIgnored(directory) || isFileContainSymbolicLinks(directory, currentDir)) {
+                            continue;
+                        }
+                        subQueue.add(directory);
+                    } else {
+                        File file = subDirOrFile;
+                        if (isFileIgnored(file)) {
+                            continue;
+                        }
+
+                        if (_config._isSearchInContent) {
+                            if (isSearchByContentIgnoredFor(file)) {
+                                continue;
+                            }
+                            List<FitFile.ContentMatchUnit> contentMatches = getContentMatches(file, _config._isOnlyFirstContentMatch);
+
+                            if (contentMatches.size() == 0) {
+                                continue;
+                            }
+
+                            String path = file.getCanonicalPath().replace(_config._rootSearchDir.getCanonicalPath() + "/", "");
+                            _result.add(new FitFile(path, false, contentMatches));
+                        }
+                    }
+
+                    if (!_config._isSearchInContent) {
+                        getFileIfNameMatches(subDirOrFile);
+                    }
+
+                    publishProgress(_currentQueueLength + subQueue.size(), _currentSearchDepth, _result.size(), _countCheckedFiles);
+                }
+            } catch (Exception ignored) {
+            }
+
+            return subQueue;
+        }
+
+
+        @Override
+        protected void onProgressUpdate(Integer... values) {
```

Comment:
Looks like we always expect exactly 4 args. Should just name them.

Reply:
this is a method from Android AsyncTask class. Signature should match

Reply:
Ah sorry, nevermind